scanpy/get.py

```diff
@@ -50,12 +51,21 @@ def rank_genes_groups_df(
     -------
     >>> import scanpy as sc
     >>> pbmc = sc.datasets.pbmc68k_reduced()
-    >>> sc.tl.rank_genes_groups(pbmc, groupby="louvain", use_raw=True, n_genes=pbmc.shape[1])
+    >>> sc.tl.rank_genes_groups(pbmc, groupby="louvain", use_raw=True)
     >>> dedf = sc.get.rank_genes_groups_df(pbmc, group="0")
     """
-    d = pd.DataFrame()
-    for k in ['scores', 'names', 'logfoldchanges', 'pvals', 'pvals_adj']:
-        d[k] = adata.uns[key][k][group]
+    if isinstance(group, str):
+        group = [group]
+    if group is None:
+        group = list(adata.uns[key]['names'].dtype.names)
+    colnames = ['names', 'scores', 'logfoldchanges', 'pvals', 'pvals_adj']
```

Comment:
Why not get it all at once?

```suggestion
    colnames = ['names', 'scores', 'logfoldchanges', 'pvals', 'pvals_adj']
    if {"pts", "pts_rest"}.issubset(adata.uns[key]):
        colnames += ["pts", "pts_rest"]
```

Reply:
I think it's not possible to get it all at once because the structure of pts and pts_rest are different than the rest of the recarrays:

<img width="1110" alt="image" src="https://user-images.githubusercontent.com/1140359/101387536-90bfd500-388c-11eb-932e-eaf826eb1410.png">

This is why I melt it and merge it after processing `['names', 'scores', 'logfoldchanges', 'pvals', 'pvals_adj']`. @Koncopd  any reason why we don't follow the same structure in pts and pts_rest?

Reply:
@gokceneraslan `pts `and `pts_rest` are not sorted and filtered by scores, this is not relevant for them. Dataframe is a more convenient and compact form for `pts` instead of recarrays.

Reply:
Ah, hadn't realized they were stored differently.

Storing things this way makes more sense to me, but it's not obvious to me from the documentation of `sc.tl.rank_genes_groups` that `pts` would be stored differently.